scripts/add-icon.js

```diff
@@ -0,0 +1,71 @@
+import fs from 'node:fs/promises';
+import inquirer from 'inquirer';
+import chalkPipe from 'chalk-pipe';
+import logSymbols from 'log-symbols';
+import getRelativeLuminance from 'get-relative-luminance';
+import { getIconsDataString, getIconDataPath, titleToSlug } from './utils.js';
+
+const hexPattern = /^#?[a-fA-F0-9]{6}$/;
+const iconsData = JSON.parse(await getIconsDataString());
+
+const prompts = [
+  {
+    type: 'input',
+    name: 'title',
+    validate: (text) => {
+      if (!text) return 'This field is required';
+      if (iconsData.icons.find((x) => x.title === text))
+        return 'This icon title already exist';
+      return true;
+    },
+    message: 'Title',
+  },
+  {
+    type: 'input',
+    name: 'hex',
+    message: 'Hex',
+    validate: (text) =>
+      hexPattern.test(text) ? true : 'This should be a 6-digit hex code',
+    transformer: (text) => {
+      const color = text.startsWith('#') ? text : `#${text}`;
+      const luminance = hexPattern.test(text)
```

Comment:
Why is the `hex` validated again? The `transformer` will not be executed if the validation fails, this is redundant.

Reply:
Because the `getRelativeLuminance` will throw an error if the value is not valid. 🤷‍♂️